.github/workflows/release.yml

```diff
@@ -0,0 +1,26 @@
+name: Release
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - 'lib/rbs_rails/version.rb'
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      id-token: write
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          bundler-cache: true
+
+      - name: Publish to RubyGems
+        uses: rubygems/release-gem@v1
```

Comment:
Note:

`rubygems/release-gem` configures "trusted publishing" credentials and releases a package via `rake release` command.

`rake release` task goes 1) package build, 2) git tag creation, and 3) gem push.

refs:

* https://github.com/rubygems/release-gem
* https://github.com/rubygems/rubygems/blob/master/bundler/lib/bundler/gem_helper.rb#L65-L69